keps/sig-cli/2368-kubectl-default-container/README.md

```diff
@@ -0,0 +1,277 @@
+
+- At least 2 release cycles pass to gather feedback and bug reports during
+- Documentations, add it to [well-known annotations docs](https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/)
+
+#### Beta -> GA Graduation
+
+- Gather feedback from developers and surveys
+- At least 2 release cycles pass to gather feedback and bug reports during
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag in 1.21
+-- As it is not compatible with `kubectl.kubernetes.io/default-logs-container` for kubectl log. Hence, this annotation will be deprecated.
+- Deprecate the flag in 1.23.
+
+
+### Upgrade / Downgrade Strategy
+None
+
+### Version Skew Strategy
+None
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism:
+      - This feature is explicitly opt-in since it need user to add specified
+        annotation in pod.
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+        - No. Disabling the feature would be a client behaviour.
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? 
+        - No. It is a client behaviour only.
+
+* **Does enabling the feature change any default behavior?**
+  - No. The old behavior is notification when no default container is specified.
+    Current behavior is cover that once specified annotation is in the pod spec.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  - Yes. When an Env 'KUBE_DEFAUTL_CONTAINER_ANNOTATION=disabled', we will ignore
+    the specified annotation. This Env is only for disabling this.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  - Nothing. Just unset Env KUBE_DEFAUTL_CONTAINER_ANNOTATION to enable it.
+
+* **Are there any tests for feature enablement/disablement?**
+  - There are unit tests in `staging/src/k8s.io/kubectl/pkg/cmd/exec/` and 
+    `staging/src/k8s.io/kubectl/pkg/polymorphichelpers/` that
+    verify the behaviour.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?  
+  - None
+* **What specific metrics should inform a rollback?
+  - None
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+  - I do not believe this is applicable.
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.? Even if applying deprecation policies, they may still surprise some users. No.
+  - `kubectl logs` use default annotation like `kubectl.kubernetes.io/default-logs-container` before 1.21 
```

Comment:
Again, but this is my opinion, I think it's too radical to deprecate this annotation :)

I guess they can co-exist and let default-logs-container be prioritized, but this is my opinion

Reply:
yes, if you do that then there is no backward compatibility issue, again assuming no automated migration of annotations